src/sage/tensor/modules/free_module_automorphism.py

```diff
@@ -1085,13 +1135,62 @@ def det(self):
             1
 
         """
-        self.matrix() # forces the update of the matrix in the module's default
-                      # basis, to make sure that the dictionary self._matrices
-                      # is not empty
-        return next(iter(self._matrices.values())).det() # pick a random value in the
-                                                # dictionary self._matrices
-                                                # and compute the determinant
+        return self._some_matrix().det
+
+    determinant = det
+
+    @lazy_attribute
+    def fcp(self):
+        r"""
+        Return the factorization of the characteristic polynomial of ``self``.
+
+        INPUT:
+
+        - ``var`` -- variable
+
+        EXAMPLES::
+
+            sage: M = FiniteRankFreeModule(QQ, 2, name='M')
+            sage: e = M.basis('e')
+            sage: a = M.automorphism([[1,1],[0,2]], name='a')
+            sage: a.matrix(e)
+            [1 1]
+            [0 2]
+            sage: a.fcp()                                                               # needs sage.libs.pari
+            (x - 2) * (x - 1)
+            sage: a.fcp('T')                                                            # needs sage.libs.pari
+            (T - 2) * (T - 1)
+        """
+        return self._some_matrix().fcp
+
+    @lazy_attribute
+    def minimal_polynomial(self):
```

Comment:
Same question as for `characteristic_polynomial`:
```suggestion
    def minimal_polynomial(self, var):
```

Reply:
No, because this is the implementation of the lazy attribute. When it is invoked, it returns the method of the matrix.

Reply:
OK, I  see. This is unfortunate regarding the documentation as shown  in the reference manual (https://deploy-preview-37826--sagemath.netlify.app/html/en/reference/tensor_free_modules/sage/tensor/modules/free_module_automorphism#sage.tensor.modules.free_module_automorphism.FreeModuleAutomorphism.minimal_polynomial) : the method `minimal_polynomial` appears as if it takes no argument.

Reply:
I agree that this is unfortunate...